src/init.cpp

```diff
@@ -1290,6 +1290,51 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     // as they would never get updated.
     if (!ignores_incoming_txs) node.fee_estimator = std::make_unique<CBlockPolicyEstimator>();
 
+    // Check port numbers
+    for (const std::string port_option : {
+        "-port",
+        "-rpcport",
+    }) {
+        if (args.IsArgSet(port_option)) {
+            const std::string port = args.GetArgs(port_option)[0];
```

Comment:
In commit "Validate `port` options" (1dae86bfd2259e4d9a9f1cd292a10cc17f626876)

Suggest replacing `args.GetArgs(port_option)[0]` with `args.GetArg(port_option)`. `IsArgSet` will true and the `[0]` subscript will be out of bounds if the argument is negated (`-noport`)

Reply:
Good catch, thanks